pybop/problems/gitt.py

```diff
@@ -0,0 +1,56 @@
+import pybop
+
+
+class GITT(pybop.FittingProblem):
```

Comment:
Yes, I deliberately sticked to the current `FittingProblem` to make it much easier to maintain. I would say that writing it as an example notebook does not reduce the maintenance as, if any changes take place, the example will break in the same way as the class will.

Regarding end users, it is always hard to judge as there as many types of users as people out there. But even a clear example might have too many moving parts to non-expert users and streamlining the process in a class which much fewer choices helps them get in (plus, you can throw in additional checks to prevent them doing some silly things). Happy to chat further about that!

Reply:
Thanks for building on the `FittingProblem` class. As above, I think the GITT problem class would be a good place to include user warnings so I would keep it separate from the example.